### PR TITLE
Fix melt front propagation laser direction

### PR DESCRIPTION
**Pull request by nmuch**
This PR fixes the domain of the melt front simulation, so that the laser heat soure acts opon the material correctly.
Fyi @mschreter

### CODE REVIEW

**Comment by mschreter** [comment]
Does this change have to do with the following line, indicating that the heated volume has to be below the applied laser heat source? If yes, I would add here a comment/assert. 

https://github.com/MeltPoolDG/MeltPoolDG/blob/32cbc019ffa0dcb2c195cbc68f3ebd65d2002473/source/heat/laser_heat_source_gusarov.cpp#L23

**Comment by nmuch** [comment]
> Does this change have to do with the following line, indicating that the heated volume has to be below the applied laser heat source? 

Correct. Otherwise, the heat source model does not make mush sense.

>If yes, I would add here a comment/assert.
> 
> https://github.com/MeltPoolDG/MeltPoolDG/blob/32cbc019ffa0dcb2c195cbc68f3ebd65d2002473/source/heat/laser_heat_source_gusarov.cpp#L23

I think an assert does not really work here, because this equation is evaluated for the entire domain which also includes the surrounding "air"-domain. 
I think the doc describes the usage of this laser model quite well
https://github.com/MeltPoolDG/MeltPoolDG/blob/32cbc019ffa0dcb2c195cbc68f3ebd65d2002473/include/meltpooldg/heat/laser_heat_source_gusarov.hpp#L19-L36
We can think about adding a coordinate system to this sketch in addition to the comment https://github.com/MeltPoolDG/MeltPoolDG/blob/32cbc019ffa0dcb2c195cbc68f3ebd65d2002473/include/meltpooldg/heat/laser_heat_source_gusarov.hpp#L54-L55

File: include/meltpooldg/simulations/melt_front_propagation/melt_front_propagation.hpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ namespace MeltPoolDG::Simulation::MeltFrontPropagation
           refinements[0] = 3;
           // create mesh
           const Point<2> left(0, 0);
-          const Point<2> right(x_max, y_max);
+          const Point<2> right(x_max, -z_max);
```
Comment (peterrum) [inline include/meltpooldg/simulations/melt_front_propagation/melt_front_propagation.hpp:107]:
```suggestion
          const Point<2> left(0, -z_max);
          const Point<2> right(x_max, 0);
```
I guess deal.II will re-orient the cells internally, but the notion is normally `left-bottom-front` and `right-top-back`.

**Comment by mschreter** [comment]
> We can think about adding a coordinate system to this sketch in addition to the comment
> https://github.com/MeltPoolDG/MeltPoolDG/blob/32cbc019ffa0dcb2c195cbc68f3ebd65d2002473/include/meltpooldg/heat/laser_heat_source_gusarov.hpp#L54-L55

This would be a good idea!

**Review by mschreter — APPROVED** [review]

File: include/meltpooldg/simulations/melt_front_propagation/melt_front_propagation.hpp (hunk 1 above)
Comment (mschreter) [reply include/meltpooldg/simulations/melt_front_propagation/melt_front_propagation.hpp]: @peterrum: Do you see an issue here?

Comment (nmuch) [reply include/meltpooldg/simulations/melt_front_propagation/melt_front_propagation.hpp]: I already applied @peterrum 's suggestion

**Pull request merged**